wtxmgr/db.go

```diff
@@ -1287,6 +1288,124 @@ func deleteRawUnminedInput(ns walletdb.ReadWriteBucket, outPointKey []byte,
 	return nil
 }
 
+// serializeLockedOutput serializes the value of a locked output.
+func serializeLockedOutput(id uint64, expiry time.Time) []byte {
+	var v [8 + 8]byte
+	byteOrder.PutUint64(v[:8], id)
+	byteOrder.PutUint64(v[8:], uint64(expiry.Unix()))
+	return v[:]
+}
+
+// deserializeLockedOutput deserializes the value of a locked output.
+func deserializeLockedOutput(v []byte) (uint64, time.Time) {
+	lockID := byteOrder.Uint64(v[:8])
+	expiry := time.Unix(int64(byteOrder.Uint64(v[8:])), 0)
+	return lockID, expiry
+}
+
+// isLockedOutput determines whether an output is locked. If it is, its assigned
+// ID is returned, along with its absolute expiration time. If the output lock
+// exists, but its expiration has been met, then it's removed.
+func isLockedOutput(ns walletdb.ReadBucket, op wire.OutPoint) (uint64,
+	time.Time, bool) {
+
+	// The bucket may not exist, indicating that no outputs have ever been
+	// locked, so we can just return now.
+	lockedOutputs := ns.NestedReadBucket(bucketLockedOutputs)
+	if lockedOutputs == nil {
+		return 0, time.Time{}, false
+	}
+
+	// Retrieve the output lock, if any, and extract the relevant fields.
+	k := canonicalOutPoint(&op.Hash, op.Index)
+	v := lockedOutputs.Get(k)
+	if v == nil {
+		return 0, time.Time{}, false
+	}
+	lockID, expiration := deserializeLockedOutput(v)
+
+	// If the output lock has already expired, delete it now.
+	if !time.Now().Before(expiration) {
```

Comment:
Use `After` here for clarity?

Reply:
`After` doesn't handle `time.Now == expiration`.